ipam/store.go

```diff
@@ -17,64 +17,119 @@ package ipam
 
 import (
 	"database/sql"
+	"errors"
+	"fmt"
 	"github.com/romana/core/common"
 	"log"
+	"strings"
 )
 
-type Vm struct {
-	//	Id        uint64 `json:"id"`
-	Ip           string         `json:"ip"`
-	TenantId     string         `json:"tenant_id"`
-	SegmentId    string         `json:"segment_id"`
-	HostId       string         `json:"host_id"`
-	Name         string         `json:"instance"`
+// Endpoint represents an endpoint (a VM, a Kubernetes Pod, etc.)
+// that is to get an IP address.
+type Endpoint struct {
+	Ip           string         `json:"ip,omitempty"`
+	TenantId     string         `json:"tenant_id,omitempty"`
+	SegmentId    string         `json:"segment_id,omitempty"`
+	HostId       string         `json:"host_id,omitempty"`
+	Name         string         `json:"name,omitempty"`
 	RequestToken sql.NullString `json:"request_token" sql:"unique"`
-	// Ordinal number of this VM in the host/tenant combination
-	Seq uint64 `json:"sequence"`
-	// Calculated effective sequence number of this VM --
+	// Ordinal number of this Endpoint in the host/tenant combination
+	Seq uint64 `json:"-"`
```

Comment:
Means omit this field if it's JSON-serialized. This is a private flag for use inside the store. Should a comment be added?